modules/ml/src/main/java/org/apache/ignite/ml/nn/MLP.java

```diff
@@ -0,0 +1,503 @@
+     *
+     * @param layerIdx Layer index.
+     * @return Weights of layer with given index.
+     */
+    public Matrix weights(int layerIdx) {
+        assert layerIdx >= 1;
+        assert layerIdx < architecture.layersCount() || below != null;
+
+        if (layerIdx < belowLayersCount())
+            return below.weights(layerIdx - architecture.layersCount());
+        else
+            return layers.get(layerIdx - belowLayersCount() - 1).weights;
+    }
+
+    /**
+     * Get biases of layer with given index. Proper indexes are in [1, layersCount).
+     *
+     * @param layerIdx Layer index.
+     * @return Biases of layer with given index.
+     */
+    public Vector biases(int layerIdx) {
+        assert layerIdx >= 0;
+        assert layerIdx < architecture.layersCount() || below != null;
+
+        if (layerIdx < belowLayersCount())
+            return below.biases(layerIdx - architecture.layersCount());
+        else
+            return layers.get(layerIdx - belowLayersCount() - 1).biases;
+    }
+
+    /**
+     * Checks if layer with given index has biases.
+     *
+     * @param layerIdx Layer index.
+     * @return Value of predicate 'layer with layerIdx has biases'.
+     */
+    public boolean hasBiases(int layerIdx) {
+        return layerIdx != 0 && biases(layerIdx) != null;
+
+    }
+
+    /**
+     * Sets the biases of layer with a given index.
+     *
+     * @param layerIdx Layer index.
+     * @param bias New values for biases.
+     * @return This MLP with updated biases.
+     */
+    public MLP setBiases(int layerIdx, Vector bias) {
+        biases(layerIdx).assign(bias);
+
+        return this;
+    }
+
+    /**
+     * Set the bias of given neuron in given layer.
+     *
+     * @param layerIdx Layer index.
+     * @param neuronIdx Neuron index.
+     * @param val New value of bias.
+     * @return This MLP with updated biases.
+     */
+    public MLP setBias(int layerIdx, int neuronIdx, double val) {
+        // Should be transformation layer.
+        assert layerIdx > 0;
+        assert architecture.transformationLayerArchitecture(layerIdx).hasBias();
+
+        biases(layerIdx).setX(neuronIdx, val);
+
+        return this;
+    }
+
+    /**
+     * Get the bias of given neuron in given layer.
+     *
+     * @param layerIdx Layer index.
+     * @param neuronIdx Neuron index.
+     * @return Bias with specified coordinates.
+     */
+    public double bias(int layerIdx, int neuronIdx) {
+        // Should be transformation layer.
+        assert layerIdx > 0;
+        assert architecture.transformationLayerArchitecture(layerIdx).hasBias();
+
+        return biases(layerIdx).getX(neuronIdx);
+    }
+
+    /**
+     * Sets the weighs of layer with a given index.
+     *
+     * @param layerIdx Layer index.
+     * @param weights New values for weights.
+     * @return This MLP with updated weights.
+     */
+    public MLP setWeights(int layerIdx, Matrix weights) {
+        weights(layerIdx).assign(weights);
```

Comment:
It's ugly - seems like get item from the set by id and after that we reassign a new set with name like method